### PR TITLE
refactor(insexpand.c): misc

### PR DESCRIPTION
**Pull request by glepnir**
has_mbyte no need as condition N times. some if replace by using ternary

### CODE REVIEW

File: src/insexpand.c [hunk 1]
```diff
@@ -553,20 +553,22 @@ ins_compl_infercase_gettext(
...
+    {
+	for (i = 0; i < char_len; ++i)
 	    wca[i] = mb_ptr2char_adv(&p);
-	else
+    }
+    else
+    {
+	for (i = 0; i < char_len; ++i)
 	    wca[i] = *(p++);
+    }
```
Comment (chrisbra) [inline src/insexpand.c:565]: that doesn't look like a clear improvement.

File: src/insexpand.c [hunk 2]
```diff
@@ -1182,10 +1186,7 @@ trigger_complete_changed_event(int cur)
     if (recursive)
 	return;
 
-    if (cur < 0)
-	item = dict_alloc();
-    else
-	item = ins_compl_dict_alloc(compl_curr_match);
+    item = cur < 0 ? dict_alloc() : ins_compl_dict_alloc(compl_curr_match);
```
Comment (chrisbra) [inline src/insexpand.c:1189]: so does this.

**Review by chrisbra — COMMENTED** [review]
Thanks. I'll include the changes minus those 2 mentioned things.

File: src/insexpand.c (hunk 1 above)
Comment (glepnir) [reply src/insexpand.c:565]: has_mbyte only needs to be checked once. Similar code can be found in find_word_end. Consider the usage scenario of has_mbyte. Maybe a macro can be used to wrap this condition?

Comment (chrisbra) [reply src/insexpand.c:565]: True, but  I found readability more important than improving the efficiency of having to check this variable only once.

**Pull request closed without merging**